server/game/player.js

```diff
@@ -896,6 +896,19 @@ class Player extends Spectator {
         });
     }
 
+    moveCardToBottomOfDeck(card, allowSave = true) {
```

Comment:
It probably should be wrapped in a `allowGameAction` for immunity purposes, like return to hand.

I would say hold off on factoring out a common method, though. Dupes + saves in general need to be reworked for issues like #1231 and #1024, which should probably be its own PR.